        <div>
        <p class="note">During program execution an error can be thrown.
            Instead of ending the program <code>an error can be caught</code> and a certain
            code can be executed.</p>

        <h2 class="h4 my-3">Error Handling</h2>
        <p>Syntax for handling error:</p>
        <textarea class="tutorial-example">try
{
    // code that may throw exception.
}
catch (ExceptionType exceptionName)
{
    // handling the error
}</textarea>
        <p>Let's look at the following example:</p>
        <textarea class="tutorial-example numeric-rows"><?php
try
{
    echo "Executing try block code.\n";

    throw new Exception("My error message.\n");
}
catch (Exception $ex)
{
    echo "Catching an error:\n";
    echo $ex->getMessage();
}</textarea>
        <p>Output:</p>
        <textarea class="tutorial-example text-only">Executing try block code.
Catching an error:
My error message.</textarea>
        <p>Since an error is thrown in the try block,
            the code from the catch block is called.</p>

        <h2 class="h4 my-3">Classes and Error Handling</h2>
        <p>An error can be thrown when a validation fails in setter.</p>
        <p>Example:</p>
        <textarea class="tutorial-example numeric-rows"><?php
class Person
{
    private $name;
    private $age;

    public function setAge($ageValue)
    {
        if ($ageValue < 0)
        {
            throw new Exception("Age cannot be less than zero.\n");
        }

        $this->age = $ageValue;
    }

    public function setName($nameValue)
    {
        $this->name = $nameValue;
    }

    public function __construct($usernameValue, $ageValue)
    {
        $this->setName($usernameValue);
        $this->setAge($ageValue);
    }
}

try
{
    $personObj = new Person("Alex", -1);
}
catch (Exception $ex)
{
    echo "Object creation failed:\n";
    echo $ex->getMessage();
}</textarea>

        <p>Output:</p>
        <textarea class="tutorial-example text-only">Object creation failed:
Age cannot be less than zero.</textarea>

        <p>At <code>line 31</code> a negative number (<code>-1</code>)
            is passed for age of the person object. This causes
            an error to be throw (due to <code>line 11</code>).</p>
    </div>